math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -149,11 +167,29 @@ public V removeVertex(int v) {
         return obj;
     }
 
+    private void cleanVertices(int v) {
+        vertices.remove(v);
+        removedVertices.remove(v);
+        if (removedVertices.contains(v - 1)) {
+            cleanVertices(v - 1);
+        }
```

Comment:
I'm not sure it is really effecient to have a recursion here where a for-loop looks OK